sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -199,7 +210,8 @@ class CodegenContext {
       case StringType => s"$input.getUTF8String($ordinal)"
       case BinaryType => s"$input.getBinary($ordinal)"
       case CalendarIntervalType => s"$input.getInterval($ordinal)"
-      case t: StructType => s"$input.getStruct($ordinal, ${t.size})"
+      case t: StructType => if (!isColumnarType(input)) { s"$input.getStruct($ordinal, ${t.size})" }
```

Comment:
While this ```getStruct()``` for ```ColumnVector``` may not be called at rutnime now, a code generator always produces two version for ```InternalRow``` and ```ColumnVector```.
Thus, this code is necessary to avoid compilation error for now. In the future, this code is necessary to correctly handle nested types.